frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -402,13 +411,15 @@ export default ({
         data: { id: message.id, emoticon }
       })
     },
-    getSimulatedState (initialize = true) {
-      return {
+    initializeState () {
+      // NOTE: this state is rendered using the chatroom contract functions
+      // so should be CAREFUL of updating the fields
+      this.messageState.contract = {
         settings: cloneDeep(this.chatRoomSettings),
         attributes: cloneDeep(this.chatRoomAttributes),
         users: cloneDeep(this.chatRoomUsers),
-        messages: initialize ? [] : this.messages,
-        onlyRenderMessage: true
+        messages: [],
+        onlyRenderMessage: true // NOTE: DO NOT CHANGE THIS, OR CHATROOM COULD BREAK
```

Comment:
Recommend `CHANGE` => `RENAME` to be more specific

Reply:
Got it.